core-web/libs/template-builder/src/lib/components/template-builder/components/add-style-classes-dialog/add-style-classes-dialog.component.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { expect } from '@jest/globals';
+import { SpectatorHost, byTestId, createHostFactory } from '@ngneat/spectator';
+
+import { HttpClientModule } from '@angular/common/http';
+import { FormsModule } from '@angular/forms';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+
+import { AutoCompleteModule } from 'primeng/autocomplete';
+import { ButtonModule } from 'primeng/button';
+import { DynamicDialogConfig, DynamicDialogRef } from 'primeng/dynamicdialog';
+
+import { AddStyleClassesDialogComponent } from './add-style-classes-dialog.component';
+import { DotAddStyleClassesDialogStore } from './store/add-style-classes-dialog.store';
+
+import { MOCK_SELECTED_STYLE_CLASSES, addStyleClassesStoreMock } from '../../utils/mocks';
+
+describe('AddStyleClassesDialogComponent', () => {
+    let component: AddStyleClassesDialogComponent;
+    let spectator: SpectatorHost<AddStyleClassesDialogComponent>;
+
+    const createHost = createHostFactory({
+        component: AddStyleClassesDialogComponent,
+        imports: [
+            AutoCompleteModule,
+            FormsModule,
+            NoopAnimationsModule,
+            ButtonModule,
+            HttpClientModule
+        ],
+        providers: [
+            {
+                provide: DynamicDialogConfig,
+                useValue: {
+                    data: {
+                        selectedClasses: MOCK_SELECTED_STYLE_CLASSES
+                    }
+                }
+            },
+            {
+                provide: DotAddStyleClassesDialogStore,
+                useValue: addStyleClassesStoreMock
+            },
+            DynamicDialogRef
+        ]
+    });
+
+    beforeEach(() => {
+        spectator = createHost(
+            '<dotcms-add-style-classes-dialog></dotcms-add-style-classes-dialog>',
+            {}
+        );
+
+        component = spectator.component;
+        spectator.detectChanges();
+
+        Object.defineProperty(window, 'matchMedia', {
+            writable: true,
+            value: jest.fn().mockImplementation((query) => ({
+                matches: false,
+                media: query,
+                onchange: null,
+                addListener: jest.fn(), // Deprecated
+                removeListener: jest.fn(), // Deprecated
+                addEventListener: jest.fn(),
+                removeEventListener: jest.fn(),
+                dispatchEvent: jest.fn()
+            }))
+        });
```

Comment:
Why do we need this? I've seen @danielmdob doing something similar, can we not dupe?